### ISSUE TITLE
Get Error "Duplicate task spec name" when flow to Multi-Instance twice

### ISSUE THREAD

**Issue by kadyang** [sartography/SpiffWorkflow#198]
Steps

- Start Process
- Complete UserTask A (Multi-Instance) 3 times (set data `approved = False`)
- exclusive gateway flow to UserTask B
- Get Error Message `KeyError: 'Duplicate task spec name: Gateway_for_task-A_start'`

Test BPMN File
```xml
<?xml version="1.0" encoding="UTF-8"?>
<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:modeler="http://camunda.org/schema/modeler/1.0" id="Definitions_0y9pl5m" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="5.0.0" modeler:executionPlatform="Camunda Platform" modeler:executionPlatformVersion="7.17.0">
  <bpmn:process id="Process_1" isExecutable="true">
    <bpmn:startEvent id="start-event" name="Start">
      <bpmn:outgoing>start-to-A</bpmn:outgoing>
    </bpmn:startEvent>
    <bpmn:sequenceFlow id="start-to-A" sourceRef="start-event" targetRef="task-A" />
    <bpmn:exclusiveGateway id="check" name="check">
      <bpmn:incoming>go-to-check</bpmn:incoming>
      <bpmn:outgoing>approved-ng</bpmn:outgoing>
      <bpmn:outgoing>approved-ok</bpmn:outgoing>
    </bpmn:exclusiveGateway>
    <bpmn:sequenceFlow id="go-to-check" sourceRef="task-A" targetRef="check" />
    <bpmn:sequenceFlow id="approved-ng" name="ng" sourceRef="check" targetRef="task-B">
      <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">approved == False</bpmn:conditionExpression>
    </bpmn:sequenceFlow>
    <bpmn:sequenceFlow id="approved-ok" name="ok" sourceRef="check" targetRef="task-C">
      <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">approved == True</bpmn:conditionExpression>
    </bpmn:sequenceFlow>
    <bpmn:endEvent id="end-event" name="End">
      <bpmn:incoming>go-to-end</bpmn:incoming>
    </bpmn:endEvent>
    <bpmn:sequenceFlow id="go-to-end" sourceRef="task-C" targetRef="end-event" />
    <bpmn:sequenceFlow id="return-A" sourceRef="task-B" targetRef="task-A" />
    <bpmn:userTask id="task-A" name="A">
      <bpmn:incoming>start-to-A</bpmn:incoming>
      <bpmn:incoming>return-A</bpmn:incoming>
      <bpmn:outgoing>go-to-check</bpmn:outgoing>
      <bpmn:multiInstanceLoopCharacteristics>
        <bpmn:loopCardinality xsi:type="bpmn:tFormalExpression">3</bpmn:loopCardinality>
      </bpmn:multiInstanceLoopCharacteristics>
    </bpmn:userTask>
    <bpmn:userTask id="task-C" name="C">
      <bpmn:incoming>approved-ok</bpmn:incoming>
      <bpmn:outgoing>go-to-end</bpmn:outgoing>
    </bpmn:userTask>
    <bpmn:userTask id="task-B" name="B">
      <bpmn:incoming>approved-ng</bpmn:incoming>
      <bpmn:outgoing>return-A</bpmn:outgoing>
    </bpmn:userTask>
  </bpmn:process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Process_1">
      <bpmndi:BPMNEdge id="Flow_1iikmgj_di" bpmnElement="start-to-A">
        <di:waypoint x="215" y="177" />
        <di:waypoint x="270" y="177" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_1a8y7jv_di" bpmnElement="go-to-check">
        <di:waypoint x="370" y="177" />
        <di:waypoint x="425" y="177" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_1bjb5x4_di" bpmnElement="approved-ng">
        <di:waypoint x="450" y="202" />
        <di:waypoint x="450" y="300" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="459" y="248" width="13" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_0lgyq5h_di" bpmnElement="approved-ok">
        <di:waypoint x="475" y="177" />
        <di:waypoint x="590" y="177" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="527" y="159" width="12" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_0x702io_di" bpmnElement="go-to-end">
        <di:waypoint x="690" y="177" />
        <di:waypoint x="782" y="177" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_0b9ygw9_di" bpmnElement="return-A">
        <di:waypoint x="400" y="340" />
        <di:waypoint x="320" y="340" />
        <di:waypoint x="320" y="217" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape id="_BPMNShape_StartEvent_2" bpmnElement="start-event">
        <dc:Bounds x="179" y="159" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="185" y="202" width="24" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Gateway_1qk7uju_di" bpmnElement="check" isMarkerVisible="true">
        <dc:Bounds x="425" y="152" width="50" height="50" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="436" y="122" width="29" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Activity_0lo4mlm_di" bpmnElement="task-A">
        <dc:Bounds x="270" y="137" width="100" height="80" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Activity_08cbyyr_di" bpmnElement="task-B">
        <dc:Bounds x="400" y="300" width="100" height="80" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Activity_1lr38ba_di" bpmnElement="task-C">
        <dc:Bounds x="590" y="137" width="100" height="80" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_1qba375_di" bpmnElement="end-event">
        <dc:Bounds x="782" y="159" width="36" height="36" />
        <bpmndi:BPMNLabel>
          <dc:Bounds x="790" y="202" width="20" height="14" />
        </bpmndi:BPMNLabel>
      </bpmndi:BPMNShape>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</bpmn:definitions>

```

**Comment by danfunk** [comment]
I can confirm this is a bug.  We will get it corrected and let you know as soon as we have a fix in Dev.  Should be able to get this into the next release.

**Comment by danfunk** [comment]
Oh, and thank you for the BPMN diagram that succinctly describes the error.  Very helpful!

**Comment by danfunk** [comment]
Not a fix, but worth noting that the diagram below does not produce this error, so we just have an issue with looping back directly into a multi instance.
![image](https://user-images.githubusercontent.com/515662/180570849-9a26304f-d72a-4486-8fdd-58ffeee10738.png)

**Comment by kadyang** [comment]
thanks you comfirm this bug.

provide more information about this :
when complete B2 and run `do_engine_steps()` to go to the B3, it will get same error.

![multi-instance](https://user-images.githubusercontent.com/24424599/180724799-5c7246be-258f-4afa-8b55-490bf4ec894b.png)

there is my code
https://github.com/kadyang/spiffworkflow-practice/tree/master/ex01

**Comment by danfunk** [comment]
Rather than add an additional task in the the loop, place a task in the original flow, and then go back to that task.  For example, place B3 BETWEEN the start event and A, don't just add additional tasks to the loop.

**Comment by danfunk** [comment]
Fixed in f0f45c9
